src/vscode-dts/vscode.proposed.textSearchProviderNew.d.ts

```diff
@@ -113,25 +113,25 @@ declare module 'vscode' {
 		/**
 		 * Options to specify the size of the result text preview.
 		 */
-		previewOptions: {
+		previewOptions?: {
 			/**
 			 * The maximum number of lines in the preview.
 			 * Only search providers that support multiline search will ever return more than one line in the match.
 			 * Defaults to 100.
 			 */
-			matchLines: number;
+			matchLines?: number;
 
 			/**
 			 * The maximum number of characters included per line.
 			 * Defaults to 10000.
 			 */
-			charsPerLine: number;
+			charsPerLine?: number;
 		};
 
 		/**
 		 * Exclude files larger than `maxFileSize` in bytes.
 		 */
-		maxFileSize: number;
```

Comment:
changing these to be optional- the provider can enforce the default value if not set